service/history/workflow/mutable_state_impl.go

```diff
@@ -938,30 +943,63 @@ func (ms *MutableStateImpl) GetStartVersion() (int64, error) {
 		if err != nil {
 			return 0, err
 		}
-		firstItem, err := versionhistory.GetFirstVersionHistoryItem(versionHistory)
-		if err != nil {
-			return 0, err
+
+		if !versionhistory.IsEmptyVersionHistory(versionHistory) {
```

Comment:
Use IsCurrentVersionHistoryEmpty?

Reply:
We still need to get the current version history (then to get the first item) if it's not empty.